pkg/services/alerting/notifiers/slack.go

```diff
@@ -246,7 +262,7 @@ func (sn *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 	}
 	imageURL := ""
 	// default to file.upload API method if a token is provided
-	if sn.Token == "" {
+	if sn.token == "" {
```

Comment:
So the thing here, is that if there's a token, then the public image is not going to appear.

This was necessary when using the old api, where a token was empty only in case that the URL was a Webhook. 

I think that we should remove that now, and on line 292:

```
if sn.NeedsImage() && imageURL != "" {
		attachment["image_url"] = imageURL
}
```

replace with:

```
if sn.NeedsImage() {
		attachment["image_url"] = evalContext.ImagePublicURL
}
```

in that case, if there's a public image url, then it will go through the request body - if not it will just be empty (no harm). Then it will check if there's an image on disk that should be uploaded and it will be posted through `file.Upload` api path. 

@aknuds1 let me know if this makes any sense!

Reply:
@dsotirakis I'm not sure if I understand what has changed, that should make these changes necessary? I have to admit that I don't know this image uploading logic, since I wasn't involved with the original code. But I also don't see what should have changed versus the old code? If the token is empty with the new code, it will also be a webhook.

Reply:
Most likely something to check in another issue, not having to do with your PR I guess. If there's a token, we'll post text and image separately which is what currently happens so 👍

Reply:
@dsotirakis yeah, I'm also unsure if it makes sense, but it sounds as if you confirm it's independent of this PR.

Reply:
Will have a final look tomorrow and let you know